core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -69,20 +108,52 @@ class KafkaRequestHandler(id: Int,
           completeShutdown()
           return
 
+        case callback: RequestChannel.CallbackRequest =>
+          try {
+            val originalRequest = callback.originalRequest
+            
+            // If we've already executed a callback for this request, reset the times and subtract the callback time from the 
+            // new dequeue time. This will allow calculation of multiple callback times.
```

Comment:
Artem requested this. See comment here. https://github.com/apache/kafka/pull/13391#discussion_r1160849589

There is currently not a way to prevent infinite callbacks.